module/VuFind/src/VuFind/Search/Blender/Params.php

```diff
@@ -0,0 +1,295 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class Params extends \VuFind\Search\Solr\Params
+{
+    /**
+     * Secondary search params
+     *
+     * @var \VuFind\Search\Base\Params
+     */
+    protected $secondaryParams;
+
+    /**
+     * Blender configuration
+     *
+     * @var \Laminas\Config\Config
+     */
+    protected $blenderConfig;
+
+    /**
+     * Blender mappings
+     *
+     * @var array
+     */
+    protected $mappings;
+
+    /**
+     * Constructor
+     *
+     * @param \VuFind\Search\Base\Options  $options         Options to use
+     * @param \VuFind\Config\PluginManager $configLoader    Config loader
+     * @param HierarchicalFacetHelper      $facetHelper     Hierarchical facet helper
+     * @param \VuFind\Search\Base\Params   $secondaryParams Secondary search params
+     * @param \Laminas\Config\Config       $blenderConfig   Blender configuration
+     * @param array                        $mappings        Blender mappings
+     */
+    public function __construct(
+        \VuFind\Search\Base\Options $options,
+        \VuFind\Config\PluginManager $configLoader,
+        HierarchicalFacetHelper $facetHelper,
+        \VuFind\Search\Base\Params $secondaryParams,
+        \Laminas\Config\Config $blenderConfig,
+        $mappings
+    ) {
+        parent::__construct(
+            $options,
+            $configLoader,
+            $facetHelper
+        );
+
+        $this->secondaryParams = $secondaryParams;
+        $this->blenderConfig = $blenderConfig;
+        $this->mappings = $mappings;
+    }
+
+    /**
+     * Pull the search parameters
+     *
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     */
+    public function initFromRequest($request)
+    {
+        parent::initFromRequest($request);
+        $this->secondaryParams->initFromRequest($this->translateRequest($request));
+    }
+
+    /**
+     * Create search backend parameters for advanced features.
+     *
+     * @return \VuFindSearch\ParamBag
+     */
+    public function getBackendParameters()
+    {
+        $params = parent::getBackendParameters();
+        if (!is_callable([$this->secondaryParams, 'getBackendParameters'])) {
+            throw new \Exception(
+                'Secondary backend missing support for getBackendParameters'
+            );
+        }
+        $secondaryParams = $this->secondaryParams->getBackendParameters();
+        $params->set(
+            'secondary_backend',
+            $secondaryParams
+        );
+        return $params;
+    }
+
+    /**
+     * Translate a request for the secondary backend
+     *
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return \Laminas\Stdlib\Parameters
+     */
+    protected function translateRequest($request)
```

Comment:
This is a super-long function; I wonder if it can be refactored (and also whether, in the long term, we can develop a mechanism for handling special cases more elegantly).

Reply:
It's now been split, but I'll look into the possibility of moving the translation to the backend code. Params is currently doing a bit too much of backendy stuff.

Reply:
Moving the translation doesn't seem feasible, it's too complicated.